### PR TITLE
[WIP] [knx] Add support for using hardware TPM modules

### PR DESCRIPTION
**Pull request by holgerfriedrich**
This as an attempt to store passwords for KNX secure protected by a TPM module.
The password is stored encrypted and can only be decrypted with this specific TPM.
The proof of concept was done on a Rapsberry PI with a LetsTrust TPM module on top.

- Add TpmInterface, a class built on top of Tss.Java. This lib is to be included in a special way due to inconsistencies in package creation which makes it incompatible to OSGI.
- Add console commands ```knx tpm-info``` and ```knx tpm-encrypt <password>```. Output to be used instead of the password. Use space in front on the knx to avoid you password to be stored to console history.

Disclaimer: Use at you own risk.

Disclaimer: Storing encrypted passwords does not bring perfect security for the password - anyone who can access your machine can use the TPM as well to decode, RPI is not a secure system, Java is not secure at all. But it is nice that you do not need to worry about disclosing you password in backups or screenshots :-)

### CODE REVIEW

**Comment by lsiepel** [comment]
@holgerfriedrich unfortunately i don;t have a `knx` system, so i can;t be much of help here. Anyway, i was wondering how you see the future for this PR. At the moment it is the olderst PR and that always gets a bit extra attention ;-)
Do we foresee this to be merged at all and/or what steps are there to be taken to get there?

**Comment by holgerfriedrich** [comment]
Thanks for asking.
This PR gets a rebase from time to time, as I use it in production :-)

Though, there are a few reasons why I did not follow up in the last months:
* base library - TSS.java - is basically unmaintained. It has been written as some kind of demo by a team at Microsoft, but does no longer get attention. My proof of concept for KNX uses RSA keys, taking up to a minute to compute. Elliptic curves would be much faster, but I could not get it running with a real TPM and TSS.java. No idea if it ever worked.
* I did not find time to base on something more stable like tpm2-tss (C library) - maybe interfacing C libraries is finally easy when we move to Java 25??
* basically this should move into core
* I have some concerns that people kill their TPMs - this is not an issue on an RPi with a separately added TPM module, but might be on a more expensive board

So I don't know where this ends....
